ClosedXML/Excel/Style/XLBorder.cs

```diff
@@ -1,80 +1,97 @@
+using System.Linq;
 
 namespace ClosedXML.Excel
 {
-    using System.Collections.Generic;
-
     internal class XLBorder : IXLBorder
     {
+        #region Static members
+        internal static XLBorderKey GenerateKey(IXLBorder defaultBorder)
+        {
+            XLBorderKey key;
+            if (defaultBorder == null)
+            {
+                key = XLBorderValue.Default.Key;
+            }
+            else if (defaultBorder is XLBorder)
+            {
+                key = (defaultBorder as XLBorder).Key;
+            }
+            else
+            {
+                key = new XLBorderKey
+                {
+                    LeftBorder = defaultBorder.LeftBorder,
+                    LeftBorderColor = defaultBorder.LeftBorderColor.Key,
+                    RightBorder = defaultBorder.RightBorder,
+                    RightBorderColor = defaultBorder.RightBorderColor.Key,
+                    TopBorder = defaultBorder.TopBorder,
+                    TopBorderColor = defaultBorder.TopBorderColor.Key,
+                    BottomBorder = defaultBorder.BottomBorder,
+                    BottomBorderColor = defaultBorder.BottomBorderColor.Key,
+                    DiagonalBorder = defaultBorder.DiagonalBorder,
+                    DiagonalBorderColor = defaultBorder.DiagonalBorderColor.Key,
+                    DiagonalUp = defaultBorder.DiagonalUp,
+                    DiagonalDown = defaultBorder.DiagonalDown,
+                };
+            }
+            return key;
+        }
+        #endregion Static members
+
+        private readonly XLStyle _style;
+
         private readonly IXLStylized _container;
-        private XLBorderStyleValues _bottomBorder;
-        private XLColor _bottomBorderColor;
-        private XLBorderStyleValues _diagonalBorder;
-        private XLColor _diagonalBorderColor;
-        private Boolean _diagonalDown;
-        private Boolean _diagonalUp;
-        private XLBorderStyleValues _leftBorder;
-        private XLColor _leftBorderColor;
-        private XLBorderStyleValues _rightBorder;
-        private XLColor _rightBorderColor;
-        private XLBorderStyleValues _topBorder;
-        private XLColor _topBorderColor;
 
-        public XLBorder() : this(null, XLWorkbook.DefaultStyle.Border)
-        {
+        private XLBorderValue _value;
+
+        internal XLBorderKey Key {
+            get { return _value.Key; }
+            private set { _value = XLBorderValue.FromKey(value); }
         }
 
-        public XLBorder(IXLStylized container, IXLBorder defaultBorder, Boolean useDefaultModify = true)
+        #region Constructors
+        /// <summary>
+        /// Create an instance of XLBorder initializing it with the specified value.
+        /// </summary>
+        /// <param name="style">Style to attach the new instance to.</param>
+        /// <param name="value">Style value to use.</param>
+        public XLBorder(IXLStylized container, XLStyle style, XLBorderValue value)
```

Comment:
`warning CS1573: Parameter 'container' has no matching param tag in the XML comment for 'XLBorder.XLBorder(IXLStylized, XLStyle, XLBorderValue)' (but other parameters do)